src/sentry/search/events/fields.py

```diff
@@ -2659,6 +2662,103 @@ def _resolve_error_handled_alias(self, _: str) -> SelectType:
             "cast", [self.column("error.handled"), "Array(Nullable(UInt8))"], ERROR_HANDLED_ALIAS
         )
 
+    def _project_threshold_multi_if_function(self) -> SelectType:
+        """Accessed by `_resolve_apdex_function` and `_resolve_count_miserable_function`,
+        this returns the right duration value (for example, lcp or duration) based
+        on project or transaction thresholds that have been configured by the user.
+        """
+
+        return Function(
+            "multiIf",
+            [
+                Function(
+                    "equals",
+                    [
+                        Function(
+                            "tupleElement",
+                            [self.resolve_field("project_threshold_config"), 1],
+                        ),
+                        "lcp",
+                    ],
+                ),
+                self.column("measurements.lcp"),
+                self.column("transaction.duration"),
+            ],
+        )
+
+    def _resolve_apdex_function(self, args: Mapping[str, str], alias: str) -> SelectType:
+        if args["satisfaction"]:
+            function_args = [self.column("transaction.duration"), int(args["satisfaction"])]
+        else:
+            function_args = [
+                self._project_threshold_multi_if_function(),
+                Function("tupleElement", [self.resolve_field("project_threshold_config"), 2]),
+            ]
+
+        return Function("apdex", function_args, alias)
+
+    def _resolve_count_miserable_function(self, args: Mapping[str, str], alias: str) -> SelectType:
+        if args["satisfaction"]:
+            lhs = self.column("transaction.duration")
+            rhs = int(args["tolerated"])
+        else:
+            lhs = self._project_threshold_multi_if_function()
+            rhs = Function(
+                "multiply",
+                [
+                    Function(
+                        "tupleElement",
+                        [self.resolve_field("project_threshold_config"), 2],
+                    ),
+                    4,
+                ],
+            )
+        col = args["column"]
+
+        return Function(
+            "uniqIf",
+            [
+                self.resolve_field_alias(col) if self.is_field_alias(col) else self.column(col),
+                Function("greater", [lhs, rhs]),
+            ],
+            alias,
+        )
```

Comment:
Can this be refactored together with the previous function? There's lots of duplication and I don't really understand what's happening.